packages/react-components/react-tabs/library/src/components/Tab/useTabAnimatedIndicator.styles.ts

```diff
@@ -67,6 +67,11 @@ const getRegisteredTabRect = (registeredTabs: Record<string, TabRegisterData>, v
   return element ? calculateTabRect(element) : undefined;
 };
 
+/**
+ * Checks if a tab value is neither null nor undefined.
+ */
+const isDefined = (value: TabValue): boolean => value !== undefined && value !== null;
```

Comment:
- no need for explicit type annotation ( not part of public API )
- can be shortened by leveraging non strict equality

```suggestion
const isDefined = (value: TabValue) => value != null;
```

Reply:
**off topic for this PR:**

going forward Im wondering about the tab value data type allowance ? `unknown` is quite generic without any constraints, like user can sepcify anything basically 

```
<Tab value={[]}>First Tab</Tab> 
<Tab value={{}}>First Tab</Tab> 
<Tab value={()=>{}}>First Tab</Tab> 
<Tab value={new Set()}>First Tab</Tab> 
// use your wildest imagination...
```

IMO we should incorporate some extent of validation for the data layer

Reply:
> **off topic for this PR:**
> 
> going forward Im wondering about the tab value data type allowance ? `unknown` is quite generic without any constraints, like user can sepcify anything basically
> 
> ```
> <Tab value={[]}>First Tab</Tab> 
> <Tab value={{}}>First Tab</Tab> 
> <Tab value={()=>{}}>First Tab</Tab> 
> <Tab value={new Set()}>First Tab</Tab> 
> // use your wildest imagination...
> ```
> 
> IMO we should incorporate some extent of validation for the data layer

I agree, however, it would likely result in a breaking change.

Reply:
> * no need for explicit type annotation ( not part of public API )
> * can be shortened by leveraging non strict equality

The `!=` operator is not allowed by ESLint; so, it may be better to leave the current code rather than using a shorthand and adding an ESLint ignore directive.

Reply:
> I agree, however, it would likely result in a breaking change.

it would be a fix not a breaking change. api surface doesn't change, only dev time validation is added. if invalid values are passed in userland ATM we ship invalid experiences.

Reply:
> is not allowed by ESLint
Not all eslint rules we have in place make sense ( they exist for historical reason / nobody cleaned things up ).
Now especially in this use0case where the usage is scoped, it is safer/ok practice to use disable pragma if needed as that would not creep in multiple places.

Last but not least, non strict equality comparison is a common known pattern => it would produce smaller JS footprint.

Reply:
> > I agree, however, it would likely result in a breaking change.
> 
> it would be a fix not a breaking change. api surface doesn't change, only dev time validation is added. if invalid values are passed in userland ATM we ship invalid experiences.

I still think that having static types and runtime validation out of sync is not the best approach. If a type is permitted by the component API, it should not generate a runtime warning or error indicating that it is not supported.

Here 

> it would be a fix not a breaking change. api surface doesn't change, only d

Created a separate issue for this topic #32425